### ISSUE TITLE
Timestamp column cannot be found when joining to a subquery

### ISSUE THREAD

**Issue by SimonPStevens** [questdb/questdb#867]
**Describe the bug**
When using a subquery in combination with a SAMPLE BY clause the timestamp column cannot be identified, causing the query to fail with the error "TIMESTAMP column is required but not provided"

**To Reproduce**
Run the following SQL to setup the tables (Taken from the first database sample in the docs here: https://questdb.io/docs/get-started/first-database)

```sql
CREATE TABLE sensors (ID LONG, make STRING, city STRING);

INSERT INTO sensors
SELECT
    x ID, --increasing integer
    rnd_str('Eberle', 'Honeywell', 'Omron', 'United Automation', 'RS Pro') make,
    rnd_str('New York', 'Miami', 'Boston', 'Chicago', 'San Francisco') city
FROM long_sequence(10000) x;

CREATE TABLE readings
AS(
    SELECT
        x ID,
        timestamp_sequence(to_timestamp('2019-10-17T00:00:00', 'yyyy-MM-ddTHH:mm:ss'), rnd_long(1,10,2) * 100000L) ts,
        rnd_double(0)*8 + 15 temp,
        rnd_long(0, 10000, 0) sensorId
    FROM long_sequence(10000000) x)
TIMESTAMP(ts)
PARTITION BY MONTH;
```
This query will error with the message "TIMESTAMP column is required but not provided":

```sql
SELECT ts, a.city, a.make, avg(temp)
FROM readings timestamp(ts)
JOIN
    (SELECT ID sensId, city, make
    FROM sensors
    WHERE city='Miami' AND make='Omron') a
ON readings.sensorId = a.sensId
WHERE ts ='2019-10-21;1d'
SAMPLE BY 1h;
```
**Workaround**
This modified query using a regular join will work successfully:

```sql
SELECT ts, city, make, avg(temp)
FROM readings 
JOIN sensors a ON readings.sensorId = a.ID
WHERE  a.city = 'Miami' AND a.make='Omron'and ts ='2019-10-21;1d'
SAMPLE BY 1h
```

**Environment:**
 - OS: Windows
 - Browser: Firefox
 - Version: 5.0.6.1

**Comment by bluestreak01** [comment]
Apologies it took a while to get to this issue! This will be fixed in 6.0.4

**Comment by bluestreak01** [comment]
Semantics of interval search changed since 6.0. It is `in` instead of `=`:

```sql
SELECT ts, a.city, a.make, avg(temp)
FROM readings timestamp(ts)
JOIN
    (SELECT ID sensId, city, make
    FROM sensors
    WHERE city='Miami' AND make='Omron') a
ON readings.sensorId = a.sensId
WHERE ts in '2019-10-21;1d'
SAMPLE BY 1h;
```
